parser/src/json/compiler.rs

```diff
@@ -568,12 +569,14 @@ impl Compiler {
             let node = self.builder.lexeme(mk_regex(regex), true);
             Ok(node)
         } else {
-            Ok(self.lexeme(&format!(
-                "\"{}{{{},{}}}\"",
-                CHAR_REGEX,
-                min_length,
-                max_length.map_or("".to_string(), |v| v.to_string())
-            )))
+            Ok(self.lexeme(
+                &format!(
+                    "(?s:.{{{},{}}})",
```

Comment:
`derivre` seems smart enough to match `\uD83D\uDCA9` with `.`, so length is counted appropriately